packages/engine-server/src/util/cache.ts

```diff
@@ -0,0 +1,23 @@
+/** Cache interface. */
+export interface Cache<K, T> {
+  /** returns data object if a the given key is found; undefined otherwise. */
+  get(key: K): T | undefined;
+
+  /** Set the given data, potentially removing a different item depending
+   *  on cache implementation and cache state. */
+  set(key: K, data: T): void;
+}
+
+/** Null object implementation of {@link Cache} to be used when we
+ *  dont want the code to actually use the cache. */
+export class NullCache<K, T> implements Cache<K, T> {
+  // @ts-ignore
```

Comment:
Unless there is a special circumstance, avoid using this statement since it skips typechecking. Alternatives to some common use cases:
- unused variables: prefix the variable with a `_`  (eg. `_foo`) to have typescript ignore it

Reply:
Sounds good.